pkg/lib/hooks.js

```diff
@@ -324,3 +324,36 @@ export function useEvent(obj, event, handler) {
 export function useInit(func, deps, comps, destroy = null) {
     return useObject(func, destroy, deps || [], comps);
 }
+
+/* - usePolkitPermissions(actions)
+*
+ * function Component(arg) {
+ *   const permissions = usePolkitPermissions([
+ *     'org.freedesktop.NetworkManager.reload',
+ *     'org.freedesktop.NetworkManager.network-control'
+ *   ]);
+ *
+ *   ...
+ *  }
+ *
+ * The returned value from the hook is an array of booleans,
+ * in the order that they were specified, specifying if the
+ * action is allowed or not.
+ * ]);
+*/
+export function usePolkitPermissions(actions) {
+    const [permissions, setPermissions] = useState([]);
+
+    useEffect(() => {
+        Promise.allSettled(actions.map((action) =>
+            // pkcheck returns a 0 status code if the check passed
+            cockpit.spawn(['sh', '-c', `pkcheck --action-id ${action} --process $$ --allow-user-interaction 2>&1`])
+        )).then((results) => {
+            setPermissions(results.map((result) =>
+                result.status === 'fulfilled'));
+        });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
OOI, what dependency does eslint complain about?

Reply:
Accident - I had it in there while testing. I will remove it!

Reply:
Actually, this is required - eslint doesn't realise that the `JSON.stringify(actions)` call is actually acting on a prop, so it complains

Reply:
Yeah, that's weird -- why does it have to be stringified in the first place? (Note: I don't know react effects well)

Reply:
The actions object is an array, and due to how `useEffect` works, it does deep equality, but sucks at doing it on arrays - see https://github.com/facebook/react/issues/14476#issuecomment-471199055 for explanation from one of the core React devs

Reply:
Thanks! Closing that thread then.